cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/etl/common/TransformExecutor.java

```diff
@@ -68,12 +69,14 @@ public TransformResponse runOneIteration(IN input) throws Exception {
     return new TransformResponse(terminalNodeEntriesMap, errors);
   }
 
-  private <T> void executeTransformation(final String stageName, Collection<T> input) throws Exception {
+  private <T> void executeTransformation(final String prevStageName, final String stageName,
```

Comment:
like mentioned earlier, we should refactor the transform executor and relevant classes so that they pass around `TransformOutput<T>` (use a better class name) that contains the output record as well as the stage that output it.